server/src/main/java/org/elasticsearch/index/mapper/DynamicTemplate.java

```diff
@@ -72,12 +73,8 @@ public static MatchType fromString(String value) {
     public enum XContentFieldType {
         OBJECT {
             @Override
-            public String defaultMappingType() {
-                return ObjectMapper.CONTENT_TYPE;
-            }
-            @Override
-            public String toString() {
-                return "object";
+            boolean supportsRuntimeField() {
```

Comment:
In hindsight, we may want to rework the DynamicRuntimeFieldsBuilder interface that plugins have to implement for the dynamic runtime mode. Given that the notion of runtime field is added to XContentFieldType, we could tie the two together, as there is now some more subtle link between the two. I will do some more thinking around this as a follow-up

Reply:
++, I was thinking as I was reading this that it would be nice to move most of `applyMatchingTemplate` into the dynamic template itself